test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp

```diff
@@ -0,0 +1,436 @@
+    EXPECT_EQ(nullptr, statistics_participant->lookup_topicdescription("INVALID_TOPIC"));
+
+    // 7. Disable statistics DataWriter
+    ret = statistics_participant->disable_statistics_datawriter(HISTORY_LATENCY_TOPIC);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_EQ(nullptr, statistics_participant->lookup_topicdescription(HISTORY_LATENCY_TOPIC));
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(PDP_PACKETS_TOPIC));
+    EXPECT_EQ(eprosima::fastdds::dds::TypeSupport(nullptr), statistics_participant->find_type(
+            history_latency_type.get_type_name()));
+
+    // 8. Enable previous statistics DataWriter with an inconsistent QoS
+    ret = statistics_participant->enable_statistics_datawriter(
+            HISTORY_LATENCY_TOPIC, inconsistent_qos);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_INCONSISTENT_POLICY);
+    EXPECT_EQ(nullptr, statistics_participant->lookup_topicdescription(HISTORY_LATENCY_TOPIC));
+    EXPECT_EQ(eprosima::fastdds::dds::TypeSupport(nullptr), statistics_participant->find_type(
+            history_latency_type.get_type_name()));
+
+    // RETCODE_BAD_PARAMETER error has precedence over RETCODE_INCONSISTENT_POLICY
+    ret = statistics_participant->enable_statistics_datawriter(
+            "INVALID_TOPIC", inconsistent_qos);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_BAD_PARAMETER);
+
+    // 9. Try to delete DomainParticipant and check that it fails because statistics DataWriters are still enabled
+    EXPECT_TRUE(eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->
+            delete_participant(statistics_participant) == 
+            eprosima::fastrtps::types::ReturnCode_t::RETCODE_PRECONDITION_NOT_MET);
+
+    // 10. Disable already disabled DataWriter
+    ret = statistics_participant->disable_statistics_datawriter(HISTORY_LATENCY_TOPIC);
+    ASSERT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+
+    // 11. Disable invalid topic name
+    ret = statistics_participant->disable_statistics_datawriter("INVALID_TOPIC");
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_BAD_PARAMETER);
+
+    // 12. Remove DDS entities
+    ret = statistics_participant->disable_statistics_datawriter(NETWORK_LATENCY_TOPIC);
+    ASSERT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_EQ(nullptr, statistics_participant->lookup_topicdescription(NETWORK_LATENCY_TOPIC));
+    EXPECT_EQ(eprosima::fastdds::dds::TypeSupport(nullptr), statistics_participant->find_type(
+            network_latency_type.get_type_name()));
+
+    ret = statistics_participant->disable_statistics_datawriter(PUBLICATION_THROUGHPUT_TOPIC);
+    ASSERT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_EQ(nullptr, statistics_participant->lookup_topicdescription(PUBLICATION_THROUGHPUT_TOPIC));
+    // The type is being used by another topic yet
```

Comment:
Then check it is still present. Same for similar comments below.

Reply:
Done in b295b5fd0a602a3075552ff51925ccfb8afa32b3